### PR TITLE
fix mouse moves over bauhaus causing histogram redraws

### PR DESCRIPTION
**Pull request by dterrahe**
As mentioned in a kindly submitted bug report (it was somewhat lengthy and I can't exactly remember where this came up) since hovering over a bauhaus widget now causes it to be highlighted (very subtly, when using the grey theme) gtk not unreasonably assumes that this could select a completely different, possibly larger, font and triggers a resizing and redrawing avalanche of the whole side panel, including the histogram. So the histogram gets redrawn multiple times using up a bunch of cpu.

This PR doesn't register the hovering state in the widget itself, but instead checks it in the drawing routine. Which seems to solve the problem for bauhaus. A similar issue within the standard gtk notebook widget (see for example filmic) can't be easily circumvented since we don't have control over that code.

@AxelG-DE mentioned an issue with flickering histograms which I assume this doesn't resolve, but it wouldn't hurt testing and then providing more details. Thanks!

### CODE REVIEW

File: src/bauhaus/bauhaus.c [hunk 1]
```diff
@@ -2177,7 +2179,8 @@ static gboolean _widget_draw(GtkWidget *widget, cairo_t *crf)
   GdkRGBA *fg_color = default_color_assign();
   GdkRGBA *bg_color;
   GdkRGBA *text_color = default_color_assign();
-  const GtkStateFlags state = gtk_widget_get_state_flags(widget);
+  const gboolean hovering = widget == darktable.bauhaus->hovered ? GTK_STATE_FLAG_PRELIGHT : 0;
```
Comment (ralfbrown) [inline src/bauhaus/bauhaus.c:2182]: Shouldn't this variable be a GtkStateFlags, since it gets or'ed with another value of that type?

**Comment by AxelG-DE** [comment]
I will test on Wednesday or latest at weekendVon meinem Mobiltelefon gesendet

**Comment by jenshannoschwalm** [comment]
Nice. Works for me.

**Comment by AxelG-DE** [comment]
> @AxelG-DE mentioned an issue with flickering histograms which I assume this doesn't resolve, but it wouldn't hurt testing and then providing more details. Thanks!

gave it a quick glance, just with regards to the flickering. It didn't change, as expected. (I was doubting even, whether or not it got a bit slower in flickering, means more inertia even, but that easily can be a misperception)

**Review by TurboGit — APPROVED** [review]
Thanks, working for me and indeed I see in the process manager that dt is not eating CPU cycles by flying over the bauhaus widgets anymore. Good finding. Definitely the 4.4 revision will be a version toward performance and fluidity of UI.

I'll merge now, just send me a little text for the release notes. TIA.

**Comment by dterrahe** [comment]
Release Note:
- a workaround was implemented for the mouse hover effect over sliders and dropdowns, that used to cause the whole side panel, including histogram, to be redrawn on each move between widgets, reducing cpu consumption.

**Comment by TurboGit** [comment]
Thanks!

**Pull request merged**